### PR TITLE
Fix masked targetci in computeCumulativeCICorrelation() (#228)

### PR DESCRIPTION
**Pull request by rdotsch**
## Summary

- `computeCumulativeCICorrelation()` returns an all-NA curve when `targetci` was generated with a mask, because `cor()` defaults to `use = "everything"` and masked pixels are `NA`. Passing `use = "complete.obs"` correlates over the unmasked pixels only.
- Unmasked targets contain no `NA`, so every existing call is bit-identical.
- The zero-variance all-NA case (responses that cancel exactly) is unaffected — `cor()` against a constant is `NA` under any `use` setting.

## Plan

`notes/plan-masked-cumcorr-228.md` — review before implementation.

## Reproducibility impact

Only the masked-targetci case changes: `NA` → real correlations over unmasked pixels. No existing unmasked call is affected.

## Test plan

- [ ] Flip the "a masked targetci makes every correlation NA" test to assert real correlations
- [ ] Verify the zero-variance all-NA test still passes unchanged
- [ ] Verify the unmasked tests still pass unchanged
- [ ] `devtools::check()` clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by rdotsch** [comment]
@codex review

File: notes/plan-masked-cumcorr-228.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+- **Unmasked targets contain no `NA`.** `complete.obs` on a pair with no missing
+  values is identical to `everything` — bit-identical output, nothing changes for
+  the common case.
+- **The zero-variance all-NA case is unaffected.** Responses that cancel exactly
+  give a constant (all-zero) CI; `cor()` against a constant is `NA` under any
+  `use` setting. That case is separately tested and documented.
+- **No new failure mode.** If the mask covers *every* pixel, `complete.obs` on an
+  all-NA pair gives `NA` (no complete cases), which is correct — there is nothing
+  to correlate.
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply notes/plan-masked-cumcorr-228.md]: Good catch — confirmed. `cor(..., use = "complete.obs")` throws `no complete element pairs` when the target is entirely `NA`. Switching to `use = "pairwise.complete.obs"`, which returns `NA` in that case. Verified all four scenarios: no NAs (identical to `everything`), partial NAs (real correlation), all NAs (`NA`, no error), zero variance (`NA` with warning). Will add a test for the fully-masked edge case.

**Comment by rdotsch** [comment]
@codex review

File: NEWS.md [hunk 2]
```diff
@@ -94,6 +94,15 @@
 
 ## Bug fixes
 
+- **`computeCumulativeCICorrelation()` now returns real correlations when `targetci` was
+  generated with a mask.** `generateCI()` stores `NA` in every pixel a `mask` excludes,
+  and the correlation was taken over all pixels (`use = "everything"`), so a single masked
+  pixel made every point on the curve `NA` however strong the signal. Correlations are now
+  computed over the unmasked pixels only. A fully masked target (every pixel `NA`) still
+  returns an all-`NA` curve — there are no complete pairs to correlate.
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply NEWS.md:109]: Added a "Reproducibility impact" paragraph to the entry: calls with a partially masked targetci now return numeric correlations where they previously returned all-NA, and code checking for NA on the curve will see a different answer.

**Comment by rdotsch** [comment]
@codex review

**Comment by rdotsch** [comment]
@codex review
